src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```
@@ -364,23 +382,28 @@ private static void SetSslCertificate(IntPtr contextPtr, SafeX509Handle certPtr,
             {
                 throw CreateSslException("Certificate pivate key check failed");
             }
-
         }
 
         private static SslException CreateSslException(string message)
         {
-            return new SslException(message);
+            ulong errorVal = libssl.ERR_get_error();
+            string msg = message + ": " + Marshal.PtrToStringAnsi(libssl.ERR_reason_error_string(errorVal));
+            return new SslException(msg, (int)errorVal);
         }
 
         private static SslException CreateSslException(string message, int error)
         {
             if (error == libssl.SslErrorCode.SSL_ERROR_SYSCALL)
             {
-                return new SslException(message);
+                return new SslException(message, error);
+            }
+            else if (error == libssl.SslErrorCode.SSL_ERROR_SSL)
+            {
+                throw Interop.libcrypto.CreateOpenSslCryptographicException();
```

Why throw rather than return?